boards/components/src/debug_queue.rs

```diff
@@ -9,41 +9,51 @@
 //! Usage
 //! -----
 //! ```rust
-//! let buf = static_init!([u8; 1024], [0; 1024]);
-//! DebugQueueComponent::new(buf).finalize(());
+//! DebugQueueComponent::new().finalize(components::debug_queue_component_static!());
 //! ```
 
 // Author: Guillaume Endignoux <guillaumee@google.com>
 // Last modified: 05 Mar 2020
 
+use core::mem::MaybeUninit;
 use kernel::collections::ring_buffer::RingBuffer;
 use kernel::component::Component;
-use kernel::static_init;
 
-pub struct DebugQueueComponent {
-    buffer: &'static mut [u8],
+#[macro_export]
+macro_rules! debug_queue_component_static {
+    () => {{
+        let ring = kernel::static_buf!(kernel::collections::ring_buffer::RingBuffer<'static, u8>);
+        let queue = kernel::static_buf!(kernel::debug::DebugQueue);
+        let wrapper = kernel::static_buf!(kernel::debug::DebugQueueWrapper);
+        let buffer = kernel::static_buf!([u8; 1024]);
```

Comment:
This change takes away the ability for boards to configure the buffer length, because the component assumes 1024

Reply:
Decided that this is fine since nothing upstream configures the length anyway, marking resolved. If some future user needs this configurability they can submit a PR implementing it.